app/webpacker/components/Results/Rankings/index.jsx

```diff
@@ -8,44 +10,85 @@ import Loading from '../../Requests/Loading';
 import ResultsFilter from '../resultsFilter';
 
-export default function Wrapper({
-  event, region, year, rankingType, gender, show,
-}) {
+const ActionTypes = {
+  SET_EVENT: 'SET_EVENT',
+  SET_REGION: 'SET_REGION',
+  SET_RANKING_TYPE: 'SET_RANKING_TYPE',
+  SET_GENDER: 'SET_GENDER',
+  SET_SHOW: 'SET_SHOW',
+};
+
+function parseInitialStateFromUrl(url) {
+  const urlPattern = /\/results\/rankings\/(\d+)\/(\w+)/; // Matches `/results/rankings/{event}/{rankingType}`
+  const match = url.match(urlPattern);
+
+  if (!match) {
+    throw new Error('URL does not match the expected pattern.');
+  }
+
+  const [, event, rankingType] = match; // Extract event and rankingType from regex groups
+
+  const urlObj = new URL(url);
+  const params = urlObj.searchParams;
+  const region = params.get('region') || 'world'; // Default to 'all' if not provided
+  const gender = params.get('gender') || 'All'; // Default to 'all' if not provided
+  const show = params.get('show') || '100 persons'; // Default to 'Persons' if not provided
+
+  return {
+    event,
+    region,
+    gender,
+    show,
+    rankingType,
+  };
+}
+
+function filterReducer(state, action) {
+  switch (action.type) {
+    case ActionTypes.SET_EVENT:
+      return { ...state, event: action.payload };
+    case ActionTypes.SET_REGION:
+      return { ...state, region: action.payload };
+    case ActionTypes.SET_RANKING_TYPE:
+      return { ...state, rankingType: action.payload };
+    case ActionTypes.SET_GENDER:
+      return { ...state, gender: action.payload };
+    case ActionTypes.SET_SHOW:
+      return { ...state, show: action.payload };
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
+  }
+}
+
+export default function Wrapper() {
   return (
     <WCAQueryClientProvider>
-      <Rankings
-        initialEvent={event}
-        initialRegion={region}
-        initialYear={year}
-        initialRankingType={rankingType}
-        initialGender={gender}
-        initialShow={show}
-      />
+      <Rankings />
     </WCAQueryClientProvider>
   );
 }
 
-export function Rankings({
-  initialEvent, initialRegion, initialRankingType, initialGender, initialShow,
-}) {
-  const [event, setEvent] = useState(initialEvent);
-  const [region, setRegion] = useState(initialRegion ?? 'all');
-  const [rankingType, setRankingType] = useState(initialRankingType);
-  const [gender, setGender] = useState(initialGender);
-  const [show, setShow] = useState(initialShow ?? 'Persons');
+export function Rankings() {
+  // Define the initial state
+  const initialState = useMemo(() => parseInitialStateFromUrl(window.location.href), []);
 
-  const filterState = useMemo(() => ({
-    event,
-    setEvent,
-    region,
-    setRegion,
-    rankingType,
-    setRankingType,
-    gender,
-    setGender,
-    show,
-    setShow,
-  }), [event, gender, rankingType, region, show]);
+  // Use the reducer
+  const [filterState, dispatch] = useReducer(filterReducer, initialState);
```

Comment:
You can remove the `initialState` const and just do
```suggestion
  const [filterState, dispatch] = useReducer(filterReducer, window.location.href, parseInitialStateFromUrl);
```
The value will only be calculated on initial render, so it's `memo`ed for free.